### DISCUSSION TITLE
Absence or not visible text at axes

### DISCUSSION THREAD

**Discussion by SuperBoss9** [OpenHantek/OpenHantek6022#204] [Q&A] [answered]
Hello Everyone,

I'm a new user of OpenHantek 6022 so don't know may be this is made by the design but I don't see the axes text (voltage for example) and the lower field values descriptions (I see only the values of some parameters).
Use windows 10 with 3.2.4 OpenHantek.
![Screenshot 2021-07-18 202608](https://user-images.githubusercontent.com/53711021/126076671-de2fefe0-d089-44e1-aeb3-601be477e651.png)

**Comment by Ho-Ro** [comment]
~Looks like a problem with OpenGL. Please do these tests:
Start the program from cmd.exe and check for error messages.
As above, start from cmd.exe, but use the command line option as described [there](https://github.com/OpenHantek/OpenHantek6022#opengl-support).
Try also option  `--useGLSL150` to force OpenGL SL version 1.50 instead of the default of 1.20.~

EDIT: wrong assumption, this is no issue.

**Reply by SuperBoss9** [reply]
1. An the result is:
OpenHantek6022 (3.2.4)
"4.6.148201 Compatibility Profile Context 20.50.02 27.20.15002.118"
Attribute Qt::AA_ShareOpenGLContexts must be set before QCoreApplication is created.
Attribute Qt::AA_UseOpenGLES must be set before QCoreApplication is created.
Graphic: 4.6.148201 Compatibility Profile Context 20.50.02 27.20.15002.118 - GLSL version 1.20
QObject::connect: Cannot queue arguments of type 'QList<double>'
(Make sure 'QList<double>' is registered using qRegisterMetaType().)
3. And result is:
OpenHantek6022 (3.2.4)
"4.6.148201 Compatibility Profile Context 20.50.02 27.20.15002.118"
Attribute Qt::AA_ShareOpenGLContexts must be set before QCoreApplication is created.
Attribute Qt::AA_UseOpenGLES must be set before QCoreApplication is created.
Graphic: 4.6.148201 Compatibility Profile Context 20.50.02 27.20.15002.118 - GLSL version 1.50
QObject::connect: Cannot queue arguments of type 'QList<double>'
(Make sure 'QList<double>' is registered using qRegisterMetaType().)
2. And the results are:
.\OpenHantek.exe --useGLES
OpenHantek6022 (3.2.4)
"4.6.148201 Compatibility Profile Context 20.50.02 27.20.15002.118"
Attribute Qt::AA_ShareOpenGLContexts must be set before QCoreApplication is created.
Attribute Qt::AA_UseOpenGLES must be set before QCoreApplication is created.
QOpenGLWidget: Failed to create context
QOpenGLWidget: Failed to create context
QObject::connect: Cannot queue arguments of type 'QList<double>'
(Make sure 'QList<double>' is registered using qRegisterMetaType().)
QOpenGLWidget: Failed to create context
QOpenGLWidget: Failed to create context
QPainter::begin: Paint device returned engine == 0, type: 1
QPainter::setRenderHint: Painter must be active to set rendering hints
QPainter::font: Painter not active
QPainter::setFont: Painter not active
qt.qpa.backingstore: composeAndFlush: QOpenGLContext creation failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed

and 

.\OpenHantek.exe --e
OpenHantek6022 (3.2.4)
"4.6.148201 Compatibility Profile Context 20.50.02 27.20.15002.118"
Attribute Qt::AA_ShareOpenGLContexts must be set before QCoreApplication is created.
Attribute Qt::AA_UseOpenGLES must be set before QCoreApplication is created.
QOpenGLWidget: Failed to create context
QOpenGLWidget: Failed to create context
QObject::connect: Cannot queue arguments of type 'QList<double>'
(Make sure 'QList<double>' is registered using qRegisterMetaType().)
QOpenGLWidget: Failed to create context
QOpenGLWidget: Failed to create context
QPainter::begin: Paint device returned engine == 0, type: 1
QPainter::setRenderHint: Painter must be active to set rendering hints
QPainter::font: Painter not active
QPainter::setFont: Painter not active
qt.qpa.backingstore: composeAndFlush: QOpenGLContext creation failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed

I'm running it at my laptop with AMD processor and Radeon graphics card. Probably do I need to install something with OGL subsystem?

**Reply by Ho-Ro** [reply]
~Some remarks:
The message `Attribute ... must be set before QCoreApplication is created.` can be ignored. The same applies to `QObject::connect: Cannot queue arguments of type 'QList' ...`.
The option `--useGLES` or `-e` for short does not work and gives this error: `QOpenGLWidget: Failed to create context`. 
As I do not use windows, I have no further clue - sorry.~

> I don't see the axes text (voltage for example) and the lower field values descriptions (I see only the values of some parameters).

EDIT: Sorry, viewing your picture above on my big screen I see now that everything is ok. Please check the manual (available from the "Help" menu) for explanation how to read the screen.

**Reply by SuperBoss9** [reply]
How it should looks like?

**Comment by Ho-Ro** [comment, marked as answer]
> I don't see the axes text (voltage for example) ...

As usual on a scope you have to look at the voltage/division, dB/division or time/division settings in the voltage, spectrum and horizontal tabs on the right or on the text on top/button of the screen, e.g. 50µs/div, the screen is 10 divisions wide and 8 divisions high, so the whole screen equates a time of 500 µs. Same goes for the voltage (100mV/div) as well as the spectrum if you will look at it (it uses Hz/div and dB/div).
Because of more than one possible trace on screen you cannot display values directly on the axes.

![image](https://user-images.githubusercontent.com/12542359/126137663-cf116486-230e-41fb-947a-74830771f9f9.png)


> ... and the lower field values descriptions (I see only the values of some parameters).

Please RTFM (read the fine manual), it's available in the help menu.

![image](https://user-images.githubusercontent.com/12542359/126135481-3adc4e13-e3f2-4da2-a80f-094942a91084.png)

![image](https://user-images.githubusercontent.com/12542359/126136097-73e4db00-4b1f-49be-9a5b-a45a60f8583c.png)
